### PR TITLE
docs/api: drop non-functioning `--all`

### PR DESCRIPTION
**Pull request by p-linnane**
`--all` is not a valid option for `brew info`

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ redirect_from:
 
 ### List metadata for all {{ site.taps.core.name }} formulae or {{ site.taps.cask.name }} casks
 
-List the `brew info --json --all` output for all current {{ site.taps.core.fullname }} formulae or {{ site.taps.cask.fullname }} casks.
+List the `brew info --json --eval-all` output for all current {{ site.taps.core.fullname }} formulae or {{ site.taps.cask.fullname }} casks.
```
Comment (MikeMcQuaid) [inline docs/api.md:14]:
```suggestion
List the `brew info --json` output for all current {{ site.taps.core.fullname }} formulae or {{ site.taps.cask.fullname }} casks.
```

Comment (p-linnane) [reply docs/api.md:14]: This returns `Error: Invalid usage: this command requires a formula or cask argument`

Comment (MikeMcQuaid) [reply docs/api.md:14]: Yeh, I don't think this needs to be an executable command and, pretty soon, it'll require a dedicated command anyway.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks @p-linnane!

**Pull request merged**